containers/src/Kokkos_UnorderedMap.hpp

```diff
@@ -901,33 +924,34 @@ class UnorderedMap {
   bool modified() const { return get_flag(modified_idx); }
 
   void set_flag(int flag) const {
-    using raw_deep_copy =
-        Kokkos::Impl::DeepCopy<typename device_type::memory_space,
-                               Kokkos::HostSpace>;
-    const int true_ = true;
-    raw_deep_copy(m_scalars.data() + flag, &true_, sizeof(int));
+    Kokkos::parallel_for(
+        "Kokkos::set flag",
+        Kokkos::RangePolicy(typename device_type::execution_space{}, flag,
+                            flag + 1),
+        Impl::FlagSetFunctor(m_scalars, true));
     Kokkos::fence(
         "Kokkos::UnorderedMap::set_flag: fence after copying flag from "
         "HostSpace");
   }
 
   void reset_flag(int flag) const {
-    using raw_deep_copy =
-        Kokkos::Impl::DeepCopy<typename device_type::memory_space,
-                               Kokkos::HostSpace>;
-    const int false_ = false;
-    raw_deep_copy(m_scalars.data() + flag, &false_, sizeof(int));
+    Kokkos::parallel_for(
+        "Kokkos::reset flag",
+        Kokkos::RangePolicy(typename device_type::execution_space{}, flag,
+                            flag + 1),
+        Impl::FlagSetFunctor(m_scalars, false));
     Kokkos::fence(
         "Kokkos::UnorderedMap::reset_flag: fence after copying flag from "
         "HostSpace");
   }
 
   bool get_flag(int flag) const {
-    using raw_deep_copy =
-        Kokkos::Impl::DeepCopy<Kokkos::HostSpace,
-                               typename device_type::memory_space>;
-    int result = false;
-    raw_deep_copy(&result, m_scalars.data() + flag, sizeof(int));
+    bool result = false;
+    Kokkos::parallel_reduce(
+        "Kokkos::get flag",
+        Kokkos::RangePolicy(typename device_type::execution_space{}, flag,
+                            flag + 1),
+        Impl::FlagGetFunctor(m_scalars), result);
```

Comment:
Why can't this be a lambda?

Reply:
Because when doing so, NVCC complains with:

```
error: The enclosing parent function ("get_flag") for an extended __host__ __device__ lambda cannot have private or protected access within its class
```

Reply:
Eventually replaced it by a call to `deep_copy` + a scalar subview.